### ISSUE TITLE
Please provide an AppImage for Linux

### ISSUE THREAD

**Issue by probonopd** [moonlight-stream/moonlight-qt#221]
Providing an [AppImage](http://appimage.org/) would have, among others, these advantages:
- Applications packaged as an AppImage can run on many distributions (including Ubuntu, Fedora, openSUSE, CentOS, elementaryOS, Linux Mint, and others)
- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary
- No root needed
- No system libraries changed
- Works out of the box, no installation of runtimes needed
- Optional desktop integration with `appimaged`
- Optional binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Can optionally GPG2-sign your AppImages (inside the file)
- Works on Live ISOs
- Can use the same AppImages when dual-booting multiple distributions
- Can be listed in the [AppImageHub](https://appimage.github.io/) central directory of available AppImages
- Can double as a self-extracting compressed archive with the `--appimage-extract` parameter
- No repositories needed. Suitable/optimized for air-gapped (offline) machines

[Here is an overview](https://appimage.github.io/apps) of projects that are already distributing upstream-provided, official AppImages.

If you have questions, AppImage developers are on #AppImage on irc.freenode.net.

**Comment by TesseractCat** [comment]
I would appreciate this as well, since flatpak requires the installation of 'org.kde.platform' which is quite large, and doesn't support themes very well, and snap has trouble working on non-systemd os's/chroots.

**Comment by kshots** [comment]
If it helps, I wrote the attached ebuild for gentoo to install the latest git head. Adapt to your own environment...
[moonlight-9999.ebuild](https://github.com/moonlight-stream/moonlight-qt/files/3220046/moonlight-9999.ebuild.txt)

**Comment by cgutman** [comment]
I added a AppImage build script in https://github.com/moonlight-stream/moonlight-qt/commit/caca491943bcc07cdbe69ffc89aa6a451aad5c8f

I'll leave this open until an official release ships with an AppImage,

**Comment by cgutman** [comment]
I got this up and running on AppVeyor with https://github.com/moonlight-stream/moonlight-qt/commit/e9f766904eeee8a568e249416d2b41e3b05342b5 however I've hit some ABI compatibility issues with libva and VA-API drivers that cause hardware acceleration to break on my Fedora 32 system.

AppImage (broken):
```
libva info: VA-API version 0.39.0
libva info: User requested driver 'i965'
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so
libva info: va_openDriver() returns -1
00:00:00 - SDL Error (0): Failed to initialize VAAPI: -1
```

Locally compiled version (working):
```
libva info: VA-API version 1.7.0
libva info: User requested driver 'i965'
libva info: Trying to open /usr/lib64/dri/i965_drv_video.so
libva info: Found init function __vaDriverInit_1_6
libva info: va_openDriver() returns 0
00:00:01 - SDL Info (0): Initialized VAAPI 1.7
```

It looks like the VA-API major version was incremented from 0 to 1, and that broke all compatibility with older VA-API drivers. That happened in [libva 2.0.0 released on October 21, 2017](https://github.com/intel/libva/releases/tag/2.0.0).

I can probably build/bundle a newer libva in the AppImage. That will break VA-API on versions of Ubuntu prior to Bionic, but I suppose it's better than breaking modern distros.

**Comment by probonopd** [comment]
Thanks @cgutman. Sounds like VA-API made a binary-breaking change. Question: if you build the newer VA-API on e.g., Xenial, why will it fail on Ubuntu prior to Bionic?

As a last resort, could you make two AppImages, one for the old and one for the new VA-API?

**Comment by cgutman** [comment]
libva 2.0+ will fail on Xenial because Xenial's VA-API drivers are VA-API version 0.x. libva 2.0+ only supports VA-API version 1.x drivers.

Just building libva wasn't enough, because the default driver path is embedded during the build and differs between distros. I had to add fallback code to try various VA-API driver paths https://github.com/moonlight-stream/moonlight-qt/commit/b3b49b440e854934547c7a8c29e224e608d0215e

VDPAU had the same problem with a hardcoded distro-specific driver path in the binary. I implemented a fallback for that https://github.com/moonlight-stream/moonlight-qt/commit/2452e80e571221d6ea22c90688f0d4bb012e51ff

I could make 2 separate AppImages, but I would like to avoid that unless there is significant demand from users on older distros. The app still works without VAAPI, just at (potentially) reduced performance on systems where alternate hardware acceleration drivers like VDPAU or NVDEC are not available.
